### PR TITLE
ATOM-16008 [Forums]: [Bug] Materials not fully loading when rendering

### PR DESCRIPTION
**Pull request by VickyAtAZ**
Added a function in RPI::StreamingImagePool to return if there is active streaming.
In TrackView's entering game mode state, add check to wait until there is no active streaming.

Signed-off-by: Tao <qingtao@amazon.com>

### CODE REVIEW

File: Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Image/StreamingImage.h [hunk 1]
```diff
@@ -113,6 +110,7 @@ namespace AZ
             RHI::ResultCode ExpandMipChain();
 
             //! Returns the most detailed mip level currently resident in memory, where a value of 0 is the highest detailed mip.
+            //! Note: this number is not
```
Comment (jiaweig-amzn) [inline Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Image/StreamingImage.h:113]: Looks like this comment is incomplete

**Comment by smurly** [comment]
Is it practical to have a unit test for this function?

File: Code/Editor/TrackView/SequenceBatchRenderDialog.cpp [hunk 2]
```diff
@@ -1007,9 +1010,16 @@ void CSequenceBatchRenderDialog::OnUpdateEnteringGameMode()
         GetIEditor()->GetMovieSystem()->Pause();
     }
     // Spend 30 frames warming up after changing to game mode.
-    else if (m_renderContext.framesSpentInCurrentPhase++ > 30)
+    else if (m_renderContext.framesSpentInCurrentPhase++ > 30 && AZ::RPI::ImageSystemInterface::Get()->GetStreamingPool())
     {
-        EnterCaptureState(CaptureState::BeginPlayingSequence);
+        const int warmUpTimeOutFrames = 500;
+        auto streamingImagePool = AZ::RPI::ImageSystemInterface::Get()->GetStreamingPool();
```
Comment (rgba16f) [inline Code/Editor/TrackView/SequenceBatchRenderDialog.cpp:1016]: Can do the retrieve of the streaming image pool and test in the if
```c++
    if(auto streamingImagePool = AZ::RPI::ImageSystemInterface::Get()->GetStreamingPool();
      m_renderContext.framesSpentInCurrentPhase++ > 30 && streamingImagePool != nullptr)
```

**Comment by VickyAtAZ** [comment]
> 
> 
> Is it practical to have a unit test for this function?

The function is associated with lower level system. Unit tests won't do much of help to validate the real functionality. I have tested it with AtomSampleViewer streaming image sample. 
I could add an control option sample which tests this function specifically.

**Comment by smurly** [comment]
> > Is it practical to have a unit test for this function?
> 
> The function is associated with lower level system. Unit tests won't do much of help to validate the real functionality. I have tested it with AtomSampleViewer streaming image sample.
> I could add an control option sample which tests this function specifically.

The ideal scenario for unit tests is lower level systems. optimally the unit test will ensure interfaces don't break and that individual modules/units of code are as intended.
AtomSampleViewer would represent a full integration test which would be costlier to run and maintain.

**Comment by VickyAtAZ** [comment]
Out of dated pull request.

**Pull request closed without merging**